tests/opwen_email_server/utils/test_email_parser.py

```diff
@@ -145,11 +169,42 @@ def assertHasCount(self, data, snippet, expected_count):
                          'Expected {} to occur {} times but got {}'.format(snippet, expected_count, actual_count))
 
     @classmethod
-    def givenTestImage(cls, content_type='image/jpeg', status=200):
-        with open(join(TEST_DATA_DIRECTORY, 'test_image.jpg'), 'rb') as image:
+    def givenTestImage(cls, content_type='image/png', status=200):
+        with open(join(TEST_DATA_DIRECTORY, 'test_image.png'), 'rb') as image:
             image_bytes = image.read()
 
-        responses.add(responses.GET, 'http://test-url.jpg',
+        responses.add(responses.GET, 'http://test-url.png',
                       headers={'Content-Type': content_type},
                       body=image_bytes,
                       status=status)
+
+
+class FormatAttachedFiles(TestCase):
+
+    def test_format_attachments_without_attachment(self):
+        input_email = {'attachments': []}
+
+        output_email = email_parser.format_attachments(input_email)
+
+        self.assertEqual(input_email, output_email)
+
+    def test_format_attachments_with_image(self):
+        input_filename = 'test_image.png'
+        input_content = _given_test_image_base64(size='large')
+        attachment = {
+            'filename': input_filename,
+            'content': input_content
+        }
+        input_email = {'attachments': [attachment]}
+
+        output_email = email_parser.format_attachments(input_email)
+        output_attachments = output_email.get('attachments', '')
+        output_filename = ''
+        output_content = ''
+        if output_attachments:
```

Comment:
You can avoid this check by simply doing the `self.assertEqual(len(output_attachments), 1)` first and then you're sure that you can access `output_attachments[0]`.